### PR TITLE
Adding instructions to run versionize-docs in cherrypick doc

### PR DESCRIPTION
**Pull request by nikhiljindal**
Ref https://github.com/kubernetes/kubernetes/issues/18861

Adding instructions to the cherrypick doc, while we fix the script.

cc @ihmccreery

### CODE REVIEW

File: docs/devel/cherry-picks.md [hunk 1]
```diff
@@ -47,6 +47,20 @@ hack/cherry_pick_pull.sh upstream/release-3.14 98765
 This will walk you through the steps to propose an automated cherry pick of pull
  #98765 for remote branch `upstream/release-3.14`.
 
+### Cherrypicking a doc change
+
+If you are cherrypicking a change which adds a doc, then you also need to run
+build/versionize-docs.sh in the release branch to versionize that doc.
+Ideally, just running cherry_pick_pull.sh should be enough, but we are not there
+yet: https://github.com/kubernetes/kubernetes/issues/18861
```
Comment (ihmccreery) [inline docs/devel/cherry-picks.md]: Nit: prefer the [#18861](https://github.com/kubernetes/kubernetes/issues/18861) syntax.

File: docs/devel/cherry-picks.md [hunk 2]
```diff
@@ -47,6 +47,20 @@ hack/cherry_pick_pull.sh upstream/release-3.14 98765
...
+
+To cherrypick PR 123456 to release-1.1, run the following commands after running hack/cherry_pick_pull.sh:
```
Comment (ihmccreery) [inline docs/devel/cherry-picks.md]: ... and before the PR is merged, right?

**Comment by ihmccreery** [comment]
A couple comments, plus:

- please enclose code snippets in `backticks`;
- the change won't pass lint, right?  If that's the case, please make a reference to it in these docs; if not, please add a linter that will yell if someone doesn't do this.

File: docs/devel/cherry-picks.md (hunk 1 above)
Comment (nikhiljindal) [inline docs/devel/cherry-picks.md]: Updated

File: docs/devel/cherry-picks.md (hunk 2 above)
Comment (nikhiljindal) [inline docs/devel/cherry-picks.md]: Added

**Comment by nikhiljindal** [comment]
Updated doc as per comments.
Re: lint: It does not complain. It is possible to cherrypick PRs without running versionize-docs today.

I hope we can fix https://github.com/kubernetes/kubernetes/issues/18861 soon!
Can you or @zmerlynn work on it? (Not sure about the holiday schedule)

**Comment by zmerlynn** [comment]
I'd rather move doc versioning out of the release infrastructure entirely, but I'm still lost on how to do that. It causes us no end to headaches like this, but it's clear because of stuff like this that it really doesn't belong at all.

**Comment by zmerlynn** [comment]
(And I'm OOO, and probably won't be able to make the requested change even after I get back.)

**Comment by nikhiljindal** [comment]
While we are fixing the issue, I think we should atleast document the current state.
@ihmccreery WDYT?

**Comment by ihmccreery** [comment]
LGTM.

**Pull request merged**